src/app/link-list/link-list.component.html

```diff
@@ -0,0 +1,12 @@
+<div class="search-bar">
+    <input #searchBox (input)="search(searchBox.value)" 
```

Comment:
Alternatively, we can use ngModel here, so you don't need to implement search() function in the controller.